src/convertToObject.js

```diff
@@ -6,7 +6,44 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString === '') {
+    return {};
+  }
+
+  const modifiedStr = sourceString
+    .trim()
+    .split(';')
+    .map((item) => {
+      return item.trim();
+    });
+
+  if (modifiedStr.length === 1) {
+    const [key, value] = sourceString.trim().split(':');
+
+    return {
+      [key]: value.trim(),
+    };
+  }
+
+  let result = {};
+
+  modifiedStr.map((item, i) => {
+    const arr = item.trim().split(':');
```

Comment:
Ensure that both `arr[0]` (key) and `arr[1]` (value) exist before attempting to trim and use them. This will prevent potential errors when splitting items that do not contain a colon.